tests/scrape.py

```diff
@@ -1173,4 +1184,29 @@ def get_all_text(elem):
     text = ''.join(elem.itertext())
     return re.sub(r'\s+', ' ', text).strip()
 
+def get_form_params(form):
+    """Get a dictionary of default values for all the form parameters.
+    If there is a <input type=file> tag, it tries to open a file object."""
+    params = {}
+    for input in form.cssselect('input'):
+        if input.get('name') and input.get('disabled') is None:
+            type = input.get('type', 'text').lower()
+            if type in ['text', 'password', 'hidden'] or (
+               type in ['checkbox', 'radio'] and input.get('checked') is not None):
+                params[input.get('name')] = input.get('value', '')
+    for select in form.cssselect('select'):
+        if select.get('disabled') is None:
+            selections = [option.get('value', '')
+                          for option in select.cssselect('option')
+                          if option.get('selected') is not None]
+            if select.get('multiple') is not None:
+                params[select.get('name')] = selections
+            elif selections:
+                params[select.get('name')] = selections[0]
+    for textarea in form.cssselect('textarea'):
+        if (textarea.get('disabled') is None and
+            textarea.get('readonly') is None):
```

Comment:
It looks readonly textareas are still part of the submitted values, per spec:
http://www.asahi-net.or.jp/~sd5a-ucd/rec-html401j/interact/forms.html#adef-readonly

Can you fix it or add a TODO to fix it later?